[Bug]: docker builds fail due to missing git

### Bug Description

Docker builds fail at the `npm install` step, both locally and in [github actions](https://github.com/NousResearch/hermes-agent/actions/runs/24306108099/job/70967509005#step:5:4401); it appears there is some dependency being installed from git.

### Steps to Reproduce

Run `docker build` / `docker compose build`

### Expected Behavior

Image is built

### Actual Behavior

```
146.6 npm ERR! npm ERR! code ENOENT
146.6 npm ERR! npm ERR! syscall spawn git
146.6 npm ERR! npm ERR! path git
146.6 npm ERR! npm ERR! errno -2
146.6 npm ERR! npm ERR! enoent An unknown git error occurred
146.6 npm ERR! npm ERR! enoent This is related to npm not being able to find a file.
```

### Affected Component

Setup / Installation

### Messaging Platform (if gateway-related)

_No response_

### Operating System

Docker

### Python Version

Python 3.13.5

### Hermes Version

Hermes Agent v0.8.0 (2026.4.8)

### Relevant Logs / Traceback

```shell
61.36 FFmpeg (playwright ffmpeg v1011) downloaded to /opt/hermes/.playwright/ffmpeg-1011
146.6 npm ERR! code 254
146.6 npm ERR! git dep preparation failed
146.6 npm ERR! command /usr/bin/node /usr/share/nodejs/npm/bin/npm-cli.js install --force --cache=/root/.npm --prefer-offline=true --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
146.6 npm ERR! npm WARN using --force Recommended protections disabled.
146.6 npm ERR! npm ERR! code ENOENT
146.6 npm ERR! npm ERR! syscall spawn git
146.6 npm ERR! npm ERR! path git
146.6 npm ERR! npm ERR! errno -2
146.6 npm ERR! npm ERR! enoent An unknown git error occurred
146.6 npm ERR! npm ERR! enoent This is related to npm not being able to find a file.
146.6 npm ERR! npm ERR! enoent 
146.6 npm ERR! 
146.6 npm ERR! npm ERR! A complete log of this run can be found in:
146.6 npm ERR! npm ERR!     /root/.npm/_logs/2026-04-12T11_19_31_897Z-debug-0.log
146.6 
146.6 npm ERR! A complete log of this run can be found in:
146.6 npm ERR!     /root/.npm/_logs/2026-04-12T11_19_30_403Z-debug-0.log
```

### Root Cause Analysis (optional)

_No response_

### Proposed Fix (optional)

add git to the container

### Are you willing to submit a PR for this?

- [x] I'd like to fix this myself and submit a PR

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,7 +12,7 @@ ENV PLAYWRIGHT_BROWSERS_PATH=/opt/hermes/.playwright
 # Install system dependencies in one layer, clear APT cache
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-        build-essential nodejs npm python3 ripgrep ffmpeg gcc python3-dev libffi-dev procps && \
+        build-essential nodejs npm python3 ripgrep ffmpeg gcc python3-dev libffi-dev procps git && \
     rm -rf /var/lib/apt/lists/*
 
 # Non-root user for runtime; UID can be overridden via HERMES_UID at runtime

